src/bootstrap/lib.rs

```diff
@@ -652,8 +654,15 @@ impl Build {
             job::setup(self);
         }
 
-        if let Subcommand::Format { check, paths } = &self.config.cmd {
-            return format::format(&builder::Builder::new(&self), *check, &paths);
+        // hardcoded subcommands
+        match &self.config.cmd {
+            Subcommand::Format { check, paths } => {
+                return format::format(&builder::Builder::new(&self), *check, &paths);
+            }
+            Subcommand::Suggest { run, .. } => {
+                return suggest::suggest(&builder::Builder::new(&self), *run);
```

Comment:
It looks like the `paths` field is never used? Can you remove it from `Subcommand::Suggest`?

Also, this doesn't download rustfmt before running suggest - we should do that here, maybe just by moving the call on line 668 to 657.